internal/testrunner/coverageoutput.go

```diff
@@ -113,6 +132,109 @@ func (c *coberturaCoverage) bytes() ([]byte, error) {
 	return buffer.Bytes(), nil
 }
 
+// Merge merges two coverage reports for a given class.
+func (c *CoberturaClass) Merge(b *CoberturaClass) error {
+	// Check preconditions: classes should be the same.
+	equal := c.Name == b.Name &&
+		c.Filename == b.Filename &&
+		len(c.Lines) == len(b.Lines) &&
+		len(c.Methods) == len(b.Methods)
+	for idx := range c.Lines {
+		equal = equal && c.Lines[idx].Number == b.Lines[idx].Number
+	}
+	for idx := range c.Methods {
+		equal = equal && c.Methods[idx].Name == b.Methods[idx].Name &&
+			len(c.Methods[idx].Lines) == len(b.Methods[idx].Lines)
+	}
+	if !equal {
+		return errors.Errorf("merging incompatible classes: %+v != %+v", *c, *b)
+	}
+	// Update methods
+	for idx := range b.Methods {
+		c.Methods[idx].Hits += b.Methods[idx].Hits
+		for l := range b.Methods[idx].Lines {
+			c.Methods[idx].Lines[l].Hits += b.Methods[idx].Lines[l].Hits
+		}
+	}
+	// Rebuild lines
+	c.Lines = nil
+	for _, m := range c.Methods {
+		c.Lines = append(c.Lines, m.Lines...)
+	}
+	return nil
+}
+
+// Merge merges two coverage reports for a given package.
+func (p *CoberturaPackage) Merge(b *CoberturaPackage) error {
+	// Merge classes
+	for _, class := range b.Classes {
+		var target *CoberturaClass
+		for _, existing := range p.Classes {
+			if existing.Name == class.Name {
+				target = existing
+				break
+			}
+		}
+		if target != nil {
+			if err := target.Merge(class); err != nil {
+				return err
+			}
+		} else {
+			p.Classes = append(p.Classes, class)
+		}
+	}
+	return nil
+}
+
+// Merge merges two coverage reports.
+func (c *CoberturaCoverage) Merge(b *CoberturaCoverage) error {
```

Comment:
Do you think that this method can be package-only visible?

Reply:
done